sematic/db/migrations/20230706150252_pre_alembic_parity.py

```diff
@@ -0,0 +1,339 @@
+                    status_history_serialization JSONB NOT NULL,
+                    created_at timestamp without time zone NOT NULL,
+                    updated_at timestamp without time zone NOT NULL,
+
+                    PRIMARY KEY(name, namespace),
+
+                    FOREIGN KEY(run_id) REFERENCES runs (id)
+                );
+                """
+            )
+
+            conn.execute("CREATE INDEX ix_jobs_run_id ON jobs_new (run_id);")
+            conn.execute("INSERT INTO jobs_new SELECT * FROM jobs;")
+            conn.execute("DROP TABLE jobs;")
+            conn.execute("ALTER TABLE jobs_new RENAME TO jobs;")
+
+            conn.execute(
+                """
+                CREATE TABLE notes_new (
+                    id character(32) NOT NULL,
+                    user_id character(32),
+                    note TEXT NOT NULL,
+                    run_id character(32) NOT NULL,
+                    root_id character(32) NOT NULL,
+                    created_at timestamp without time zone NOT NULL,
+                    updated_at timestamp without time zone NOT NULL,
+
+                    PRIMARY KEY(id),
+
+                    FOREIGN KEY(run_id) REFERENCES runs (id),
+                    FOREIGN KEY(root_id) REFERENCES runs (id),
+                    FOREIGN KEY(user_id) REFERENCES users (id)
+                );
+                """
+            )
+            conn.execute("INSERT INTO notes_new SELECT * FROM notes;")
+            conn.execute("DROP TABLE notes;")
+            conn.execute("ALTER TABLE notes_new RENAME TO notes;")
+
+            conn.execute(
+                """
+                CREATE TABLE edges_new (
+                    id character(32) NOT NULL,
+                    source_run_id character(32),
+                    source_name TEXT,
+                    destination_run_id character(32),
+                    destination_name TEXT,
+                    artifact_id character(40),
+                    parent_id character(32),
+                    created_at timestamp NOT NULL,
+                    updated_at timestamp NOT NULL,
+
+                    PRIMARY KEY (id),
+
+                    FOREIGN KEY(artifact_id) REFERENCES artifacts (id),
+                    FOREIGN KEY(parent_id) REFERENCES edges (id),
+                    FOREIGN KEY(destination_run_id) REFERENCES runs (id),
+                    FOREIGN KEY(source_run_id) REFERENCES runs (id)
+                );
+                """
+            )
+            conn.execute("INSERT INTO edges_new SELECT * FROM edges;")
+            conn.execute("DROP TABLE edges;")
+            conn.execute(
+                "CREATE INDEX ix_edges_source_run_id ON edges_new (source_run_id);"
+            )
+            conn.execute(
+                "CREATE INDEX ix_edges_destination_run_id ON edges_new (destination_run_id);"
+            )
+            conn.execute("ALTER TABLE edges_new RENAME TO edges;")
```

Comment:
Does the index get automatically updated when the table gets renamed?

Reply:
moved to after the final rename, just to be safe.